ironfist/src/cpp/combat/creatures.cpp

```diff
@@ -137,6 +150,31 @@ void LoadCreatures() {
 						}
 				}
 
+				for (int i = 0; i < 6; i++) {
+					gMonSecondaryCost[id][i][0] = 0;
+					gMonSecondaryCost[id][i][1] = 0;
+				}
+				bool secondary_cost = false;
+				for (creature_t::secondary_cost_iterator i = c.secondary_cost().begin();
+				     i != c.secondary_cost().end();
+					 ++i) {
+					    for (int k = 0; k < ELEMENTS_IN(secondaryCostIdxTable); k++) {
+						   if (strcmp(secondaryCostIdxTable[k].resource, i->resource().c_str())
+							   == 0) {
+							   if (!secondary_cost) {
+								   gMonSecondaryCost[id][k][0] = 1;
+								   gMonSecondaryCost[id][k][1] = i->cost();
+								   secondary_cost = true;
+							   }
+							   else {
+								   EarlyShutdown("Startup Error", 
+									   "Error loading creatures.xml.\nToo many secondary resource costs per creature!");

```

Comment:
Oh cool; EarlyShutdown is definitely  a better choice than DisplayError. Thanks!

It would be nice to also give the name of the creature. We've adopted the new std::string requirements, so this would mean using std::string for the string manipulation, and overloading EarlyShutdown to also accept an std::string .